src/common/users.ts

```diff
@@ -546,8 +546,9 @@ export const shouldAllowRestore = async (
 ) => {
   const { userId, lastViewAt: lastViewAtDb } = streak;
   const user = await con.getRepository(DbUser).findOneBy({ id: userId });
-  const today = new Date();
-  const lastView = new Date(lastViewAtDb);
+  const timezone = user.timezone || 'utc';
+  const today = utcToZonedTime(new Date(), timezone);
+  const lastView = utcToZonedTime(new Date(lastViewAtDb), timezone);
```

Comment:
`lastViewAtDb` is not `Date` already?

Reply:
No, this one came from CDC. So either a number or a string. You never know :hide_the_pain: